### ISSUE TITLE
Android profiling crashes Podcast app on startup

### ISSUE THREAD

**Issue by eerhardt** [dotnet/runtime#64458]
Following the steps outlined in https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/tracing.md, and running the https://github.com/microsoft/dotnet-podcasts app, I'm seeing the app crash on startup.

Looking into the logcat, I see the following:

```
--------- beginning of crash
01-28 10:13:46.870  6563  6563 F libc    : Fatal signal 11 (SIGSEGV), code 128 (SI_KERNEL), fault addr 0x0 in tid 6563 (etConf2021.Maui), pid 6563 (etConf2021.Maui)
01-28 10:13:46.881   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
01-28 10:13:46.883   323   323 W EmuHWC2 : No layers, exit, buffer 0x7adf85b824f0
01-28 10:13:46.893  6593  6593 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstone
01-28 10:13:46.893   294   294 I tombstoned: received crash request for pid 6563
01-28 10:13:46.894  6593  6593 I crash_dump64: performing dump of process 6563 (target tid = 6563)
01-28 10:13:46.896  6593  6593 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
01-28 10:13:46.896  6593  6593 F DEBUG   : Build fingerprint: 'google/sdk_gphone_x86_64_arm64/generic_x86_64_arm64:11/RSR1.210722.013/7800151:userdebug/dev-keys'
01-28 10:13:46.896  6593  6593 F DEBUG   : Revision: '0'
01-28 10:13:46.896  6593  6593 F DEBUG   : ABI: 'x86_64'
01-28 10:13:46.897   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
01-28 10:13:46.898  6593  6593 F DEBUG   : Timestamp: 2022-01-28 10:13:46-0600
01-28 10:13:46.898  6593  6593 F DEBUG   : pid: 6563, tid: 6563, name: etConf2021.Maui  >>> com.Microsoft.NetConf2021.Maui <<<
01-28 10:13:46.898  6593  6593 F DEBUG   : uid: 10168
01-28 10:13:46.898  6593  6593 F DEBUG   : signal 11 (SIGSEGV), code 128 (SI_KERNEL), fault addr 0x0
01-28 10:13:46.898  6593  6593 F DEBUG   :     rax 0000000000000000  rbx 0000000000000000  rcx 0000000000000003  rdx 0000000000000000
01-28 10:13:46.898  6593  6593 F DEBUG   :     r8  0000000000000000  r9  0000000000000000  r10 0000000000000000  r11 0000000000000001
01-28 10:13:46.898  6593  6593 F DEBUG   :     r12 0000000000000000  r13 0000000000000000  r14 37fd000000022103  r15 0000000000000000
01-28 10:13:46.898  6593  6593 F DEBUG   :     rdi 37fd000000022103  rsi ffffffffffffffff
01-28 10:13:46.898  6593  6593 F DEBUG   :     rbp 0000000000000000  rsp 00007ffdf381b488  rip 0000789c92858e71
01-28 10:13:46.899   323   323 W EmuHWC2 : No layers, exit, buffer 0x7adf85b82c40
01-28 10:13:46.914   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
01-28 10:13:46.916   323   323 W EmuHWC2 : No layers, exit, buffer 0x7adf85b824f0
01-28 10:13:46.932   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
01-28 10:13:46.933   323   323 W EmuHWC2 : No layers, exit, buffer 0x7adf85b82c40
01-28 10:13:46.947   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
01-28 10:13:46.949   323   323 W EmuHWC2 : No layers, exit, buffer 0x7adf85b824f0
01-28 10:13:46.963   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
01-28 10:13:46.966   323   323 W EmuHWC2 : No layers, exit, buffer 0x7adf85b82c40
01-28 10:13:46.981   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
01-28 10:13:46.983   323   323 W EmuHWC2 : No layers, exit, buffer 0x7adf85b824f0
01-28 10:13:46.998   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
01-28 10:13:47.000  6593  6593 F DEBUG   : backtrace:
01-28 10:13:47.000  6593  6593 F DEBUG   :       #00 pc 0000000000053e71  /apex/com.android.runtime/lib64/bionic/libc.so (strlen+17) (BuildId: 3707c39fc397eeaa328142d90b50a973)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #01 pc 0000000000031e00  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-component-diagnostics_tracing.so (offset 0x11f000) (BuildId: 8f1859d94c179b3c1d6a09b2227284a0b92db024)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #02 pc 0000000000031f22  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-component-diagnostics_tracing.so (offset 0x11f000) (BuildId: 8f1859d94c179b3c1d6a09b2227284a0b92db024)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #03 pc 000000000001cd0b  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-component-diagnostics_tracing.so (offset 0x11f000) (BuildId: 8f1859d94c179b3c1d6a09b2227284a0b92db024)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #04 pc 00000000000200ba  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-component-diagnostics_tracing.so (offset 0x11f000) (BuildId: 8f1859d94c179b3c1d6a09b2227284a0b92db024)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #05 pc 0000000000016228  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-component-diagnostics_tracing.so (offset 0x11f000) (BuildId: 8f1859d94c179b3c1d6a09b2227284a0b92db024)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #06 pc 00000000000a5e2c  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmonosgen-2.0.so (offset 0x1dc000) (BuildId: 481e76080501014266bcc0e2d51e0eee5bc07a8c)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #07 pc 000000000006da59  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmonosgen-2.0.so (offset 0x1dc000) (BuildId: 481e76080501014266bcc0e2d51e0eee5bc07a8c)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #08 pc 000000000006d690  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmonosgen-2.0.so (offset 0x1dc000) (mono_image_open_from_data_internal+704) (BuildId: 481e76080501014266bcc0e2d51e0eee5bc07a8c)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #09 pc 000000000006dc70  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmonosgen-2.0.so (offset 0x1dc000) (mono_image_open_from_data_alc+96) (BuildId: 481e76080501014266bcc0e2d51e0eee5bc07a8c)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #10 pc 000000000001f9f1  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-android.release.so (offset 0x17a000) (xamarin::android::internal::EmbeddedAssemblies::open_from_bundles(_MonoAssemblyName*, std::__ndk1::function<_MonoImage* (unsigned char*, unsigned long, char const*)>, bool)+7169) (BuildId: 5e799662f649a18c530a65faa1a3091403366ed3)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #11 pc 000000000001fdaa  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-android.release.so (offset 0x17a000) (xamarin::android::internal::EmbeddedAssemblies::open_from_bundles(void*, _MonoAssemblyName*, char**, void*, _MonoError*)+74) (BuildId: 5e799662f649a18c530a65faa1a3091403366ed3)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #12 pc 0000000000036b97  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmonosgen-2.0.so (offset 0x1dc000) (BuildId: 481e76080501014266bcc0e2d51e0eee5bc07a8c)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #13 pc 0000000000033a60  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmonosgen-2.0.so (offset 0x1dc000) (mono_assembly_request_byname+688) (BuildId: 481e76080501014266bcc0e2d51e0eee5bc07a8c)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #14 pc 0000000000036fea  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmonosgen-2.0.so (offset 0x1dc000) (mono_assembly_load_full_alc+122) (BuildId: 481e76080501014266bcc0e2d51e0eee5bc07a8c)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #15 pc 000000000002c2f9  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-android.release.so (offset 0x17a000) (xamarin::android::Util::monodroid_load_assembly(void*, char const*)+57) (BuildId: 5e799662f649a18c530a65faa1a3091403366ed3)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #16 pc 000000000002424a  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-android.release.so (offset 0x17a000) (xamarin::android::internal::MonodroidRuntime::init_android_runtime(_JNIEnv*, _jclass*, _jobject*)+602) (BuildId: 5e799662f649a18c530a65faa1a3091403366ed3)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #17 pc 0000000000024f87  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-android.release.so (offset 0x17a000) (xamarin::android::internal::MonodroidRuntime::create_and_initialize_domain(_JNIEnv*, _jclass*, xamarin::android::jstring_array_wrapper&, xamarin::android::jstring_array_wrapper&, _jobjectArray*, xamarin::android::jstring_array_wrapper&, _jobject*, bool, bool, bool)+183) (BuildId: 5e799662f649a18c530a65faa1a3091403366ed3)
01-28 10:13:47.000  6593  6593 F DEBUG   :       #18 pc 0000000000026a34  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-android.release.so (offset 0x17a000) (xamarin::android::internal::MonodroidRuntime::Java_mono_android_Runtime_initInternal(_JNIEnv*, _jclass*, _jstring*, _jobjectArray*, _jstring*, _jobjectArray*, _jobject*, _jobjectArray*, int, unsigned char, unsigned char)+5684) (BuildId: 5e799662f649a18c530a65faa1a3091403366ed3)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #19 pc 0000000000026f2b  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/split_config.x86_64.apk!libmono-android.release.so (offset 0x17a000) (Java_mono_android_Runtime_initInternal+75) (BuildId: 5e799662f649a18c530a65faa1a3091403366ed3)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #20 pc 0000000000020de5  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/oat/x86_64/base.odex (art_jni_trampoline+437)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #21 pc 0000000000170d13  /apex/com.android.art/lib64/libart.so (BuildId: 7fbaf2a1a3317bd634b00eb90e32291e)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #22 pc 00000000004550b2  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/oat/x86_64/base.vdex (mono.MonoPackageManager.LoadApplication+342)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #23 pc 00000000001700d1  /apex/com.android.art/lib64/libart.so (BuildId: 7fbaf2a1a3317bd634b00eb90e32291e)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #24 pc 000000000045521a  /data/app/~~-rB_QpYnvDJtsQGP8OMvTA==/com.Microsoft.NetConf2021.Maui-FD1WdEA2jbSL-_iG2StZrA==/oat/x86_64/base.vdex (mono.MonoRuntimeProvider.attachInfo+82)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #25 pc 00000000004257a8  /system/framework/x86_64/boot-framework.oat (android.app.ActivityThread.installProvider+1448) (BuildId: 9ae0dca73129fa9275af95cba6a7cdde25868e76)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #26 pc 0000000000425074  /system/framework/x86_64/boot-framework.oat (android.app.ActivityThread.installContentProviders+260) (BuildId: 9ae0dca73129fa9275af95cba6a7cdde25868e76)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #27 pc 000000000041f3dd  /system/framework/x86_64/boot-framework.oat (android.app.ActivityThread.handleBindApplication+6845) (BuildId: 9ae0dca73129fa9275af95cba6a7cdde25868e76)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #28 pc 000000000041358a  /system/framework/x86_64/boot-framework.oat (android.app.ActivityThread$H.handleMessage+5786) (BuildId: 9ae0dca73129fa9275af95cba6a7cdde25868e76)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #29 pc 0000000000672267  /system/framework/x86_64/boot-framework.oat (android.os.Handler.dispatchMessage+183) (BuildId: 9ae0dca73129fa9275af95cba6a7cdde25868e76)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #30 pc 0000000000675629  /system/framework/x86_64/boot-framework.oat (android.os.Looper.loop+1433) (BuildId: 9ae0dca73129fa9275af95cba6a7cdde25868e76)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #31 pc 0000000000426da5  /system/framework/x86_64/boot-framework.oat (android.app.ActivityThread.main+661) (BuildId: 9ae0dca73129fa9275af95cba6a7cdde25868e76)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #32 pc 0000000000179016  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+806) (BuildId: 7fbaf2a1a3317bd634b00eb90e32291e)
01-28 10:13:47.001   323   323 W EmuHWC2 : No layers, exit, buffer 0x7adf85b82c40
01-28 10:13:47.001  6593  6593 F DEBUG   :       #33 pc 000000000020baa1  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+257) (BuildId: 7fbaf2a1a3317bd634b00eb90e32291e)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #34 pc 000000000062bdb6  /apex/com.android.art/lib64/libart.so (art::InvokeMethod(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jobject*, _jobject*, unsigned long)+5494) (BuildId: 7fbaf2a1a3317bd634b00eb90e32291e)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #35 pc 000000000058a298  /apex/com.android.art/lib64/libart.so (art::Method_invoke(_JNIEnv*, _jobject*, _jobject*, _jobjectArray*)+56) (BuildId: 7fbaf2a1a3317bd634b00eb90e32291e)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #36 pc 000000000008cffe  /apex/com.android.art/javalib/x86_64/boot.oat (art_jni_trampoline+222) (BuildId: 063a75624879be7d4274ea340748b04abbd86b8d)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #37 pc 00000000008de27d  /system/framework/x86_64/boot-framework.oat (com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run+141) (BuildId: 9ae0dca73129fa9275af95cba6a7cdde25868e76)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #38 pc 00000000008e6dd7  /system/framework/x86_64/boot-framework.oat (com.android.internal.os.ZygoteInit.main+2039) (BuildId: 9ae0dca73129fa9275af95cba6a7cdde25868e76)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #39 pc 0000000000179016  /apex/com.android.art/lib64/libart.so (art_quick_invoke_static_stub+806) (BuildId: 7fbaf2a1a3317bd634b00eb90e32291e)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #40 pc 000000000020baa1  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+257) (BuildId: 7fbaf2a1a3317bd634b00eb90e32291e)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #41 pc 0000000000629591  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeWithVarArgs<art::ArtMethod*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, art::ArtMethod*, __va_list_tag*)+465) (BuildId: 7fbaf2a1a3317bd634b00eb90e32291e)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #42 pc 00000000004deb49  /apex/com.android.art/lib64/libart.so (art::JNI<true>::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, __va_list_tag*)+665) (BuildId: 7fbaf2a1a3317bd634b00eb90e32291e)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #43 pc 00000000000ad0e4  /system/lib64/libandroid_runtime.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+164) (BuildId: 84eb9c8bad06a5ac4720d16d40f66380)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #44 pc 00000000000b5591  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+865) (BuildId: 84eb9c8bad06a5ac4720d16d40f66380)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #45 pc 0000000000004160  /system/bin/app_process64 (main+1616) (BuildId: ce3cc65283f87cf0181925988a809dbd)
01-28 10:13:47.001  6593  6593 F DEBUG   :       #46 pc 00000000000507d8  /apex/com.android.runtime/lib64/bionic/libc.so (__libc_init+104) (BuildId: 3707c39fc397eeaa328142d90b50a973)
01-28 10:13:47.013   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
01-28 10:13:47.015   323   323 W EmuHWC2 : No layers, exit, buffer 0x7adf85b824f0
01-28 10:13:47.030   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
01-28 10:13:47.032   323   323 W EmuHWC2 : No layers, exit, buffer 0x7adf85b82c40
01-28 10:13:47.047   323   323 W EmuHWC2 : validate: layer 25 CompositionType 1, fallback
```

### Notes

1. I am able to profile a `dotnet new maui` app just fine. There has to be something special about this app that is causing it to crash. I can share an `.apk` file, if someone is interested.

2. The `dotnet-podcasts` app is still on Maui 12. I have a branch with it updated to the latest here: https://github.com/microsoft/dotnet-podcasts/compare/main...eerhardt:UpdateToLatestMaui. If someone is trying to reproduce this locally with the latest bits, it would be easiest to start with that branch, and not `main` of `dotnet-podcasts`.

cc @steveisok @lateralusX

**Comment by eerhardt** [comment]
I'm now seeing this same crash in a `dotnet new maui` app, but deployed as a `Release` build of the app. So maybe it isn't special to the dotnet-podcast app, but being a `Release` build of a maui app.

**Comment by lateralusX** [comment]
OK, thanks, that scenario have been run in the past, so might be a due to a regression in the branches used by MAUI. I will investigate.

**Comment by lateralusX** [comment]
I have an idea that I'm looking into, we also have this https://github.com/xamarin/xamarin-android/issues/6700, so it appears there is a regression in this space, compared to previous releases.

If it's related to the use of embedded assemblies, then it is possible to turn of the image/assembly load events to get around the issue.

**Comment by lateralusX** [comment]
So you can workaround the issue by disable load event's (module, assembly), that is done when you startup a trace session by providing the following argument to dotnet-trace:

`--providers Microsoft-Windows-DotNETRuntime:0x00000014C14FCCB4:4`

The loader events uses 0x10 in that bit mask.

or you can just run with the sample profiler:

`--providers Microsoft-DotNETCore-SampleProfiler:0x0000F00000000000:4`

I verified it on a sample MAUI release compiled app and was able to extract startup nettrace files.

So this happens on release build MAUI since Xamarin Android bundles the assemblies into the image and load them a little differently during runtime initialization, looks like there is a missing field in that case that cause issues when these events try to extract that data.

I will look into a fix, but until that lands, you should be able to continue running with above workaround.

**Comment by jeromelaban** [comment]
Thanks a lot @lateralusX I'm unblocked !

**Comment by jeromelaban** [comment]
@lateralusX I'm unblocked by the trace being generated, though now it's the speedscope file that is empty (even if the trace is full). Should I open another issue for this ?

```json
{"exporter": "Microsoft.Diagnostics.Tracing.TraceEvent@2.0.64.0", 
"name": "uno-app-trace.speedscope", 
"activeProfileIndex": 0, 
"$schema": "https://www.speedscope.app/file-format-schema.json",
 "shared": { "frames": [ ] }, "profiles": [ ] }
```

**Comment by lateralusX** [comment]
@jeromelaban  Yes you need to have at least the sample profiler enabled or you won't get needed sample events that speedscope uses, in the workaround I included how you enable them both.

**Comment by eerhardt** [comment]
Thanks @lateralusX. I can confirm that workaround unblocked me as well.

**Comment by lateralusX** [comment]
Fix in dotnet/runtime main: https://github.com/dotnet/runtime/pull/64908
Fix in dotnet/runtime release-6.0: https://github.com/dotnet/runtime/pull/64911
